library/src/main/java/com/woxthebox/draglistview/swipe/ListSwipeItem.java

```diff
@@ -166,26 +173,41 @@ void setSwipeTranslationX(float x) {
         mSwipeTranslationX = Math.min(mSwipeTranslationX, getMeasuredWidth());
         mSwipeTranslationX = Math.max(mSwipeTranslationX, -getMeasuredWidth());
 
-        mSwipeView.setTranslationX(mSwipeTranslationX);
-        if (mSwipeListener != null) {
-            mSwipeListener.onItemSwiping(this, mSwipeTranslationX);
+        if (mSwipeInStyle != SwipeInStyle.SLIDE) {
+            executeTransactionX();
         }
 
         if (mSwipeTranslationX < 0) {
             if (mSwipeInStyle == SwipeInStyle.SLIDE) {
-                mRightView.setTranslationX(getMeasuredWidth() + mSwipeTranslationX);
+                float translationX = getMeasuredWidth() + mSwipeTranslationX;
+                if (mMaximumRightTranslationX == null || translationX > mMaximumRightTranslationX) {
+                    mRightView.setTranslationX(translationX);
+                    executeTransactionX();
+                }
             }
             mRightView.setVisibility(View.VISIBLE);
             mLeftView.setVisibility(View.INVISIBLE);
         } else if (mSwipeTranslationX > 0) {
             if (mSwipeInStyle == SwipeInStyle.SLIDE) {
-                mLeftView.setTranslationX(-getMeasuredWidth() + mSwipeTranslationX);
+                float translationX = -getMeasuredWidth() + mSwipeTranslationX;
+                if (mMaximumLeftTranslationX == null || translationX < -mMaximumLeftTranslationX) {
+                    mLeftView.setTranslationX(translationX);
+                    executeTransactionX();
+                }
             }
             mLeftView.setVisibility(View.VISIBLE);
             mRightView.setVisibility(View.INVISIBLE);
         } else {
             mRightView.setVisibility(View.INVISIBLE);
             mLeftView.setVisibility(View.INVISIBLE);
+            executeTransactionX();
```

Comment:
executeTransactionX(); could be called the second time here if SwipeInStyle is not slide which will cause a second call to onItemSwiping()

Maybe the execute call has to be done in an else case to the mSwipeInStyle == SwipeInStyle.SLIDE ifs or maybe make a variable like shouldExecute = false which is set to true in the different if cases and then checked after the if cases.